src/medaka-tools.cpp

```diff
@@ -0,0 +1,52 @@
+#include "medaka-tools.h"
+
+#include <array>
+#include <iostream>
+#include <dlfcn.h>
+#include <MGConfItem>
+
+#include <QTime>
+#include <QDate>
+#include <QFile>
+
+#include<chrono>
+#include<thread>
+
+using SpiMsg = std::array<uint8_t, 7>;
+
+static void Write(const SpiMsg& data) {
+	QFile multiSensorFile("/dev/MultiSensors_CD_01");
+	if(!multiSensorFile.open(QIODevice::WriteOnly | QIODevice::Text))
+	{
+		qCritical("Unable to open file for write. Check permissions");
+		return;
+	}
+	multiSensorFile.write((const char *)data.data(), data.size());
+	multiSensorFile.close();
+}
+
+static void KoiSet12H(bool value) {
+	Write({0xFE,0x01,0x01,value,0x00,0x00,0x00});
+}
+
+namespace AsteroidOS::LCD_Tools::Medaka {
+
+void SyncTime(int) {
+	KoiSet12H(!MGConfItem("/org/asteroidos/settings/use-12h-format").value().toBool());
+	QDate dateNow = QDate::currentDate();
+	QTime timeNow = QTime::currentTime();
+	Write({
+	(uint8_t) 0x50,
+	(uint8_t) 0,
+	(uint8_t) ((((dateNow.year() - 2000) & 0xEF) << 1) | ((dateNow.month() & 0x08) >> 3)),
+	(uint8_t) ((dateNow.month() << 5) | (dateNow.day() & 0x1F)),
+	(uint8_t) (((dateNow.dayOfWeek() % 7) << 5) | (timeNow.hour() & 0x1F)), //weeks are counted from sunday, sunday encoded as zero.
+	(uint8_t) (timeNow.minute() & 0xFF),
+	(uint8_t) ((timeNow.second()) & 0xFF)
```

Comment:
This line appears to be the only one that's different compared to `koi`. Was this on purpose?

Reply:
yes. medaka's display updates faster, so the 1s offset isn't relevant